addon/templates/components/ember-sparkles/pie-chart.hbs

```diff
@@ -0,0 +1,110 @@
+          (d3-style 'fill' (pipe (d3-get 'data.key') colorScale))
+          (d3-attr 'opacity' 1)
+        )
+        update=(pipe
+          (d3-style 'fill' 'purple')
+          (if with-transition (d3-transition transition) (d3-noop))
+          (d3-attr 'transform' (i 'translate(${w},${h})' h=constructors.centerHeight w=constructors.centerWidth))
+          (d3-attr 'd' constructors.arc)
+          (d3-style 'fill' (pipe (d3-get 'data.key') colorScale))
+        )
+        exit=(pipe
+          (if with-transition (d3-transition transition) (d3-noop))
+          (d3-attr 'opacity' 0)
+          (d3-remove)
+        )
+    ) d3el)}}
+
+    {{shhh (compute
+      (d3-join '.percentage' (compute constructors.pie data)
+        enter=(pipe
+          (d3-append 'text')
+          (d3-attr 'class' 'percentage')
+          (d3-attr 'transform' (ember-sparkles/arc-translate constructors.arc.centroid height=constructors.centerHeight width=constructors.centerWidth))
+          (if with-transition (d3-transition transition) (d3-noop))
+          (d3-text (d3-get 'data.percentage'))
+          (d3-attr 'opacity' 1)
+        )
+        update=(pipe
+          (if with-transition (d3-transition transition) (d3-noop))
+          (d3-attr 'transform' (ember-sparkles/arc-translate constructors.arc.centroid height=constructors.centerHeight width=constructors.centerWidth))
+          (d3-text (d3-get 'data.percentage'))
+        )
+        exit=(pipe
+          (if with-transition (d3-transition transition) (d3-noop))
+          (d3-remove)
+          (d3-attr 'opacity' 0)
+        )
+    ) d3el)}}
+
+  {{/with}}
+
+  {{#with (compute (pipe
+      (d3-append 'g')
+      (d3-attr 'class' 'legend')
+      (d3-attr 'opacity' 0)
+      (d3-attr 'transform' (i 'translate(${w}, 18)' w=(mult width (div 99 100))))
+      (d3-attr 'opacity' 1)
+  ) d3el) as |legend|}}
+
+    {{shhh (compute
+      (d3-join 'circle' domain
+        enter=(pipe
+          (d3-append 'circle')
+          (d3-style 'fill' colorScale)
+          (if with-transition (d3-transition transition) (d3-noop))
+          (d3-attr 'transform' (i 'translate(${w}, 18)' w=(sub width 18)))
+          (d3-attr 'r' (div radius 25))
+          (d3-attr 'transform' transformLegend)
+        )
+        update=(pipe
+          (if with-transition (d3-transition transition) (d3-noop))
+          (d3-style 'fill' colorScale)
+        )
+        exit=(pipe
+          (if with-transition (d3-transition transition) (d3-noop))
+          (d3-attr 'transform' (i 'translate(${w},18)' w=(sub width 18)))
+          (d3-remove)
+        )
+    ) legend)}}
+
+    {{shhh (compute (d3-join 'text' domain
+        enter=(pipe
+          (d3-append 'text')
+          (d3-attr 'opacity' 0)
+          (d3-attr 'dx' '-3.8em')
+          (d3-attr 'dy' '.35em')
+          (d3-style 'font-size' '10px')
+          (if with-transition (d3-transition transition) (d3-noop))
+          (d3-text (d3-get))
```

Comment:
I think I get what it's doing. `d3-get` should be 2 separate helpers. One that return the value and the other that gets the value. I added `r/get` helper to handle the 2nd use case. I'm going to make another helper called `r/param` which you can use to take a take a param. We should use that here instead.